client/webserver/site/src/js/wallets.ts

```diff
@@ -288,6 +292,88 @@ export default class WalletsPage extends BasePage {
       })
   }
 
+  // stepSend makes a request to get an estimated fee and displays the confirm
+  // send form.
+  async stepSend () {
+    const page = this.page
+    page.vSendFee.textContent = '0.00000000'
+    page.vSendDestinationAmt.textContent = '0.00000000'
+    page.vTotalSend.textContent = '0.00000000'
+    page.vSendErr.textContent = ''
+    page.vSendAddr.textContent = ''
+    const assetID = parseInt(page.sendForm.dataset.assetID || '')
+    const subtract = page.subtractCheckBox.checked || false
+    const conversionFactor = app().unitInfo(assetID).conventional.conversionFactor
+    const value = Math.round(parseFloat(page.sendAmt.value || '') * conversionFactor)
+    // We don't want to be on the confirm page without at least an address.
+    if (page.sendAddr.value === '') {
+      page.sendErr.textContent = 'address field cannot be empty'
+      return
+    }
```

Comment:
I know that this solution is based on my own comments, but there something that comes up a little short for me. I think what we really wanted to get tx fee estimates to display more detailed information **on the send form**, so that the user can make more informed decisions **without encountering errors**. In that case, **we really need a maximum withdraw estimate**, which tells us the maximum amount that the user can withdraw and the fees reserved in such a situation. 

The estimate for the specified amount could also be displayed **live** on the form (my intention in previous comments). Even without the max withdraw estimate, that would enable us to inform the user if the amount they've typed is invalid, solving our larger issue. But it'd be better if we could just tell them how much they can send.

I do think that displaying the information on a confirmation form like you have here is a **great** new feature, so we should get this in.

Reply:
> In that case, we really need a maximum withdraw estimate, which tells us the maximum amount that the user can withdraw and the fees reserved in such a situation.

Sure, especially for ETH and get rid of the message on the send form?

> The estimate for the specified amount could also be displayed live on the form 

Do you mean the estimate for the amount the user specified that is already shown on the confirmation page?

Reply:
Let's not do that. Just return a boolean regardless of whether they provide an address or a partial address or an empty string. Just tell the caller if its valid, and let them decide how to use the info.

Reply:
The user will want different info at different times.

1) Before they enter any information, they'll benefit from knowing what their **maximum send amount** is
2) Once they enter a value, even before hitting submit, we can display estimates for the **specified amount**. You could even wait until the confirmation form for this **IF** step 1 is in place to prevent errors on submission.
3) Once they enter an address, a small indicator (I'm thinking little green checkmark) could indicate that the **address is valid**.

Really, address validation could be a separate function, but we keep blowing up the `asset.Wallet` API when it should be solidifying, so I'm hesitant to add yet another method.

Reply:
> Once they enter an address, a small indicator (I'm thinking little green checkmark) could indicate that the address is valid.

If we want this, having an address validation endpoint won't hurt.